app/src/main/java/com/poupa/vinylmusicplayer/misc/queue/StaticPlayingQueue.java

```diff
@@ -0,0 +1,458 @@
+            originalQueue.get(i).index = i;
+
+            if (i != position && queue.get(i).index >= previousPosition) {
+                queue.get(i).index = queue.get(i).index + 1;
+            }
+        }
+
+        if (position <= this.currentPosition) {
+            this.currentPosition++;
+        }
+
+        songsIsStale = true;
+    }
+
+    // add songs after and including position, numbering need to be redone for every song after this position (+number of song)
+    public void addAllAfter(int position, @NonNull List<Song> songs) {
+        if (position >= queue.size()) {
+            position = queue.size() - 1;
+        }
+        int previousPosition = queue.get(position).index + 1;
+        position = position + 1;
+
+        int n = songs.size() - 1;
+        for (int i = n; i >= 0; i--) {
+            originalQueue.add(previousPosition, new IndexedSong(songs.get(i), previousPosition + i));
+
+            queue.add(position, new IndexedSong(songs.get(i), previousPosition + i));
+
+            if (position < this.currentPosition) {
+                this.currentPosition++;
+            }
+        }
+
+        for (int i = 0; i < queue.size(); i++) {
+            originalQueue.get(i).index = i;
+            if (!(i >= position && i <= position+n) && queue.get(i).index >= previousPosition  ) {
+                queue.get(i).index = queue.get(i).index + n + 1;
+            }
+        }
+
+        songsIsStale = true;
+    }
+
+    // move song from from to to, position are conserved
+    public void move(int from, int to) {
+        if (from == to) return;
+        final int currentPosition = this.currentPosition;
+
+        IndexedSong songToMove = queue.remove(from);
+        queue.add(to, songToMove);
+
+        if (shuffleMode == SHUFFLE_MODE_NONE) {
+            IndexedSong previousSongToMove = originalQueue.remove(from);
+            originalQueue.add(to, previousSongToMove);
+
+            for (int i = 0; i < queue.size(); i++) {
+                queue.get(i).index = i;
+                originalQueue.get(i).index = i;
+            }
+        }
+
+        if (from > currentPosition && to <= currentPosition) {
+            this.currentPosition = currentPosition + 1;
+        } else if (from < currentPosition && to >= currentPosition) {
+            this.currentPosition = currentPosition - 1;
+        } else if (from == currentPosition) {
+            this.currentPosition = to;
+        }
+
+        songsIsStale = true;
+    }
+
+    private int rePosition(int deletedPosition) {
+        int position = this.currentPosition;
+
+        if (deletedPosition < position) {
+            this.currentPosition = position - 1;
+        } else if (deletedPosition == position) { //the current position was deleted
+            if (queue.size() > deletedPosition) {
+                return position;
+            } else {
+                return position - 1;
+            }
+        }
+
+        return INVALID_POSITION;
```

Comment:
If I understand this correctly, you may not delete a position greater than the current position?